### ISSUE TITLE
fused batch norm delay_updates confusion

### ISSUE THREAD

**Issue by tdeboissiere** [tensorflow/tensorflow#7840]
In https://github.com/tensorflow/tensorflow/blob/r1.0/tensorflow/contrib/layers/python/layers/layers.py around line 347:

From its definition, `delay_updates` should delay updates to moving mean and moving average if we are training. 

The way it is defined now, when `is_training` evaluates to Trure, `smart_cond` will pick the `_delay_updates` function which actually adds the update ops to the graph. Is that intended ?

**Comment by zhangyaobit** [comment]
Yes, it is intended. Note that the ops are added to updates_collections=ops.GraphKeys.UPDATE_OPS, and will not be updated immediately, but later (why it is  called "delayed") before the loss computation [here](https://github.com/tensorflow/tensorflow/blob/r1.0/tensorflow/contrib/training/python/training/training.py#L414).

On the other hand, the _force_update will immediately update the moving mean and var, noting the added [control dependencies](https://github.com/tensorflow/tensorflow/blob/r1.0/tensorflow/contrib/layers/python/layers/layers.py#L333).

**Comment by tdeboissiere** [comment]
Thanks for the clarification. What are the benefits of delaying the update ?

**Comment by zhangyaobit** [comment]
You are welcome! Looks like it is a more efficient option in a distributed setting:

" One can set updates_collections=None to force the updates in place, but that
  can have speed penalty, specially in distributed settings."

https://github.com/tensorflow/tensorflow/blob/r1.0/tensorflow/contrib/layers/python/layers/layers.py#L399

BTW, I'm closing this issue as it has been resolved.
